### ISSUE TITLE
Training codes for Joint optimization of successive frames

### ISSUE THREAD

**Issue by hytu99** [Orange-OpenSource/Cool-Chic#23]
Hi,
Thank you for your continuous efforts in updating your project! I'm very interested in your recent work, "Improved Encoding for Overfitted Video Codecs." In this paper, the joint optimization of successive frames significantly enhances coding efficiency. However, I find it challenging to implement this training process within the current training framework.

Do you have any plans to release the training code for the joint optimization of successive frames? I would greatly appreciate any insights or guidance you could provide.

**Comment by theoladune** [comment]
Hi 👋 

Thanks for your interest in our work!

Yes, the joint encoding of several frames is one of the recent contributions to Cool-chic that we plan to implement in the coming months. As you rightly pointed out, joint RD cost optimization is not straightforward to integrate into current Cool-chic 4.0 architecture, justifying a slight delay for its release :wink: 

One simple implementation for the joint encoding would be to have a list of ``FrameEncoder``, one per frame. Then a training iteration becomes a forward pass through all the ``FrameEncoder`` and the loss is the sum of the rates and distortions of all frames. One key element to make everything converge properly is to pre-trained the ``FrameEncoder`` corresponding to the intra frames so that they output "normal" decoded images that can directly be used as reference for the remaining inter frames.

Théo
